File: BinarySearch.c
#include<stdio.h>

int binartSearch(int arr[],int n,int x)
{
    int left,mid,right;
    left=0;
    right=n-1;

    while(left<=right)
    {
        mid=(left+right)/2;

        if(arr[mid]==x)
        {
            return mid;
        }
        else if(arr[mid]>x)
        {
            right=mid-1;
        }
        else
        {
            left=mid+1;
        }
    }

    return -1;
}

int main()
{
    int n,x,i;
    int arr[100];
    printf("Enter size of array:\n");
    scanf("%d",&n);
    printf("Enter Array element:\n");

    for(i=0; i<n; i++)
    {
        scanf("%d",&arr[i]);
    }
    printf("Enter your Search Item:\n");
    scanf("%d",&x);

    printf("Your Search Result is:\n");
    int result=binartSearch(arr,n,x);
    if(result!=-1)
    {
        printf("%d",arr[result]);
    }
    else
    {
        printf("Not found");
    }

    /*
    Binary Search algorithm Time complexity :O(log2^n) and Space complexity O(1);


    */
}

File: kotlin/src/binarySearch.kt

fun binarySearch(arr:Array<Int>,key:Int):Int?
{

    var l=0
    var h=arr.size-1
    while (l<=h){
        var mid:Int=(l+h)/2

        when {
            key==arr[mid] -> return 1

            key>arr[mid] -> l=mid+1

            else -> h=mid-1
        }

    }


    return null
}
//must have a sorted array
fun main(args:Array<String>){
    print("Enter your Array size:")
    var n= readLine()!!.toInt()

    var array=Array(n){0}
    for(i in 0 until n)
        array[i]= readLine()!!.toInt()

    print("Which number you want search? :")
    var key= readLine()!!.toInt()
    if(binarySearch(array,key)!=null)
        print("Found $key")
    else
        print("Not found")

    /*
    * Must have sorted array  for binary search
    * Time Complexity O(log2^n)
    * space complexity O(1)
    * */
}